x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/index/IndexResolver.java

```diff
@@ -216,7 +216,7 @@ public void resolveNames(
 
         String[] indexWildcards = Strings.commaDelimitedListToStringArray(indexWildcard);
         Set<IndexInfo> indexInfos = new HashSet<>();
-        if (retrieveAliases) {
+        if (retrieveAliases && (clusterWildcard == null || simpleMatch(clusterWildcard, clusterName))) {
```

Comment:
This check - `clusterWildcard == null || simpleMatch(clusterWildcard, clusterName)` - is being used in another method in the same class. Maybe extract it as a separate method?

Reply:
Makes sense, extracted.